storage/internal/benchmarks/main.go

```diff
@@ -286,16 +292,20 @@ func (br *benchmarkResult) selectParams(opts benchmarkOptions) {
 		}
 
 		br.params = randomizedParams{
-			appBufferSize: opts.readQuantum * randomInt(opts.minReadSize/opts.readQuantum, opts.maxReadSize/opts.readQuantum),
+			appBufferSize: opts.readBufferSize,
 			chunkSize:     -1,    // not used for reads
 			crc32cEnabled: true,  // crc32c is always verified in the Go GCS library
 			md5Enabled:    false, // we only need one integrity validation
 			api:           api,
 			rangeOffset:   randomInt64(opts.minReadOffset, opts.maxReadOffset),
 		}
 
-		if opts.useDefaults {
-			br.params.appBufferSize = -1 // use -1 to indicate default; if we give it a value any change to defaults would not be reflected
+		if !opts.allowCustomClient {
+			br.params.appBufferSize = 4000 // default for HTTP
+
+			if api == grpcAPI {
+				br.params.appBufferSize = 32000 // default for GRPC
```

Comment:
Can you explain why this is needed?

Reply:
This sets the values that are reported later. I suppose we could remove that validation and continue to report -1 or move this into the output function